src/main/java/seedu/address/logic/commands/FindCommand.java

```diff
@@ -55,13 +71,24 @@ public CommandResult execute(Model model) {
         Predicate<Person> combinedPredicate = predicates.stream()
                 .reduce(Predicate::and)
                 .orElse(t -> false);
+
+        if (isChained) {
+            // We HAVE to copy the list here, or it will have been emptied by the time Predicate.test() is called
+            Predicate<Person> inFilteredListPredicate = new InFilteredListPredicate(
+                    new ArrayList<>(model.getFilteredPersonList()));
+            combinedPredicate = combinedPredicate.and(inFilteredListPredicate);
+        }
```

Comment:
I can somehow get what you are trying to do here but I am still a bit confused. In my understanding, you are constructing `combinedPredicate` through different methods based on whether the find command is chained or not, so it is like a if-else relationship. However, the code here seems to suggest that by default the `combinedPredicate` will be constructed in one way and only through another way if the find is specified as chained. Though both ways can work in this scenario, would it be better to show this implementation using if-else? So people can know that these two operations are if-else relationship instead of first depending on the second? Because I was really confused by this when first seeing this, thank you!

Reply:
The reason I didn't completely separate the logic for chained and non-chained predicate construction is because they differ only in one predicate (`InFilteredListPredicate`) while all other predicates are for user-specified search conditions which need to be added regardless of whether chained or not. Based on your suggestion we can rewrite like this but will introduce a bit of code duplication. Do you prefer the following or the original implementation?

```java
if (isChained) {
    combinedPredicate = predicates.stream()
            .reduce(Predicate::and)
            .orElse(t -> false);
    Predicate<Person> inFilteredListPredicate = new InFilteredListPredicate(
            new ArrayList<>(model.getFilteredPersonList()));
    combinedPredicate = combinedPredicate.and(inFilteredListPredicate);
} else {
    combinedPredicate = predicates.stream()
            .reduce(Predicate::and)
            .orElse(t -> false);
}
```

Reply:
Ah I get what you mean, sure if that's the case then there's no need to change like this, you can stick to the original approach, thank you for the detailed explaination!